functions/Get-DbaUserPermission.ps1

```diff
@@ -215,17 +215,65 @@ function Get-DbaUserPermission {
-                Write-Message -Level Verbose -Message "Processing $db on $instance"
+            $sqlFile = Join-Path -Path $script:PSModuleRoot -ChildPath "bin\stig.sql"
+            $sql = [System.IO.File]::ReadAllText("$sqlFile")
+            $sql = $sql.Replace("<TARGETDB>", $master.Name)
 
-                $db.ExecuteNonQuery($endSQL)
+            try {
+                $master.ExecuteNonQuery($sql)
+            } catch {
+                # here to avoid an empty catch
+                $null = 1
+            } # sometimes it complains about not being able to drop the stig schema if the person Ctrl-C'd before.
 
-                if ($db.IsAccessible -eq $false) {
-                    Stop-Function -Message "The database $db is not accessible" -Continue
+            try {
+                $serverDT = $master.Query($serverSQL)
+
+                foreach ($row in $serverDT) {
+                    [PSCustomObject]@{
+                        ComputerName       = $server.ComputerName
+                        InstanceName       = $server.ServiceName
+                        SqlInstance        = $server.DomainInstanceName
+                        Object             = 'SERVER'
+                        Type               = $row.Type
+                        Member             = $row.Member
+                        RoleSecurableClass = $row.'Role/Securable/Class'
+                        SchemaOwner        = $row.'Schema/Owner'
+                        Securable          = $row.Securable
+                        GranteeType        = $row.'Grantee Type'
+                        Grantee            = $row.Grantee
+                        Permission         = $row.Permission
+                        State              = $row.State
+                        Grantor            = $row.Grantor
+                        GrantorType        = $row.'Grantor Type'
+                        SourceView         = $row.'Source View'
+                    }
                 }
+            } catch {
+                Write-Message -Level warning "An error occured while retrieving server permissions."
+            }
+
+            #delete objects from running server permissions query
+            Write-Message -Level Verbose -Message "Deleting objects"
+            try {
+                $tempdb.ExecuteNonQuery($endSQL)
+            } catch {
+                # here to avoid an empty catch
+                $null = 1
+            }
+
+            # now that we have server level permissions get any database level perms
+            foreach ($db in $dbs | Where-Object { $_.isAccessible -eq $true }) {
+                Write-Message -Level warning -Message "Database $db on $instance is not accessible and will be skipped."
+            }
+
+            foreach ($db in $dbs | Where-Object { $_.isAccessible -eq $true }) {
+                Write-Message -Level Verbose -Message "Processing $db on $instance"
+
+                $db.ExecuteNonQuery($endSQL)
```

Comment:
Why do you do that (again)?

Reply:
Asked and answered.

Reply:
Executing $endSQL in a database context other than tempdb does: Nothing - because there is (or should be) no schema STIG.